internal-dns/src/dns_server.rs

```diff
@@ -134,66 +134,72 @@ async fn handle_req<'a, 'b, 'c>(
             }
         };
 
-    match record {
-        DnsRecord::AAAA(addr) => {
-            let mut aaaa = Record::new();
-            aaaa.set_name(name)
-                .set_rr_type(RecordType::AAAA)
-                .set_data(Some(RData::AAAA(addr)));
-
-            let mresp = rb.build(header, vec![&aaaa], vec![], vec![], vec![]);
+    if records.is_empty() {
+        error!(log, "No records found for {}", key);
+        nack(&log, &mr, &socket, &header, &src).await;
```

Comment:
I think we should probably return an NXDOMAIN here. The check at line 83 has passed, so the query is for a record in the zone managed by the internal DNS server, but here we have found that the record set for the queried name is empty, so we can assert that the queried name is not a thing.

Reply:
Done - refactored the "return NXDOMAIN" function, and calling it here.

Also added a test case for this behavior.